### PR TITLE
Add translation feature

### PR DESCRIPTION
**Pull request by maggiekoesno**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ target/
 **/.classpath
 **/.factorypath
 **/.project
+portfolio/src/creds/
```
Comment (dhritiwasan14) [inline .gitignore:7]: What was this folder for?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -119,16 +121,50 @@ function createReplyFormElement(parentId){
     return replyForm;
 }
 
-function createParagraphElement(className, text){
-    var paragraph; var node;
+function createParagraphElement(className, text, id=-1){
+    var paragraph; var node; var paragraphId;
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/script.js:125]: Small nit: can you make this - var paragraph, node, paragraphId; ?

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -125,4 +125,17 @@ div.subcontent{
 
 p.credit{
     font-size: xx-small;
+}
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/style.css:128]: I noticed this file is getting fairly big... consider splitting the styles into separate files.

File: .gitignore (hunk 1 above)
Comment (maggiekoesno) [reply .gitignore:7]: authentication for dev work

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -138,4 +174,23 @@ function createInputElement(name, placeholder, size, maxLength, required) {
...
+    params.append('text', text);
+    params.append('languageCode', languageCode);
+
+    fetch('/translate', {
+        method: 'POST',
+        body: params
+    }).then(response => response.text())
+    .then((translatedMessage) => {
+        textContainer.innerText = translatedMessage;
+    });
```
Comment (dhritiwasan14) [inline portfolio/src/main/webapp/script.js:195]: can this be formatted better?

**Pull request merged**